databuilder/databuilder/extractor/es_base_extractor.py

```diff
@@ -47,13 +58,43 @@ def _get_indexes(self) -> Dict:
     def _get_index_mapping_properties(self, index: Dict) -> Optional[Dict]:
         mappings = index.get('mappings', dict())
 
+        # Mapping types were removed in Elasticsearch 7. As a result, index mappings are formatted differently.
+        # See https://www.elastic.co/guide/en/elasticsearch/reference/current/removal-of-types.html
+        version = self._get_es_version()
+
         try:
-            properties = list(mappings.values())[0].get('properties', dict())
+            if int(version.split('.')[0]) >= 7:
+                properties = mappings.get('properties', dict())
+            else:
+                properties = list(mappings.values())[0].get('properties', dict())
         except Exception:
             properties = dict()
 
         return properties
 
+    def _get_attributes(self,
+                        input_mapping: Dict,
+                        parent_col_name: bool = False,
+                        separator: str = '.') -> List[ColumnMetadata]:
+        cols: List[ColumnMetadata] = []
+
+        for col_name, col_mapping in input_mapping.items():
+            qualified_col_name = str(parent_col_name) + separator + col_name if parent_col_name else col_name
+            if isinstance(col_mapping, dict):
+                if col_mapping.__contains__('properties'):
+                    # Need to recurse
+                    inner_mapping: Dict = col_mapping.get('properties', {})
+                    cols.extend(self._get_attributes(input_mapping=inner_mapping,
+                                                     parent_col_name=qualified_col_name,
```

Comment:
signature of _get_attributes expects boolean value in parent_col_name but you pass string here. can you correct the signature?

Reply:
Good catch. I will fix this.